src/uris/deepLinks/deepLinkService.ts

```diff
@@ -182,13 +232,83 @@ export class DeepLinkService implements Disposable {
 						break;
 					}
 
-					// TODO@ramint Instead of erroring, prompt the user to clone or add the repo, wait for the response,
-					// and then choose an action based on whether the repo is successfully cloned/added, of the user
-					// cancels, or if there is an error.
-					action = DeepLinkServiceAction.DeepLinkErrored;
-					message = 'No matching repo found.';
+					repoOpenAction = await this.showOpenRepoPrompt();
+					if (repoOpenAction === DeepLinkRepoOpenAction.Cancel) {
+						action = DeepLinkServiceAction.DeepLinkCancelled;
+						break;
+					}
+
+					// TODO@ramint Add cloning
+					repoOpenUri = (
+						await window.showOpenDialog({
+							title: 'Open Repository for Link',
+							canSelectFiles: false,
+							canSelectFolders: true,
+							canSelectMany: false,
+						})
+					)?.[0];
+
+					if (!repoOpenUri) {
+						action = DeepLinkServiceAction.DeepLinkCancelled;
+						break;
+					}
+
+					if (
+						repoOpenAction === DeepLinkRepoOpenAction.AddToWorkspace &&
+						(workspace.workspaceFolders?.length || 0) > 1
+					) {
+						action = DeepLinkServiceAction.OpenRepo;
+					} else {
+						// Deep link will resolve in a different service instance
+						await this.container.storage.store('deepLinks:pending', {
+							state: this._context.state,
+							action: DeepLinkServiceAction.OpenRepo,
+							uri: this._context.uri,
+							repoId: this._context.repoId,
+							remoteUrl: this._context.remoteUrl,
+							targetType: this._context.targetType,
+							targetId: this._context.targetId,
+						});
+						action = DeepLinkServiceAction.DeepLinkCancelled;
+					}
+
+					openWorkspace(repoOpenUri, {
+						location: deepLinkRepoOpenActionToOpenWorkspaceLocation[repoOpenAction],
+					});
 					break;
 
+				case DeepLinkServiceState.OpeningRepo:
+					queueMicrotask(
+						() =>
+							void window.withProgress(
+								{
+									cancellable: true,
+									location: ProgressLocation.Notification,
+									title: `Opening repo for link: ${uri}`,
```

Comment:
Let's say `repository`

Reply:
Done in commit: [Improvements from feedback](https://github.com/gitkraken/vscode-gitlens/pull/2503/commits/9e01ae826ec49b794057aa9570ec3df54ef0042d)